build.gradle

```diff
@@ -46,8 +48,15 @@ subprojects {
     apply plugin: 'org.jetbrains.dokka-android'
 
     // Enable Kotlin warnings as errors for all modules
-    tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).all {
-        kotlinOptions.allWarningsAsErrors = true
+    def enableWarningsAsErrors = rootProject.ext.getProperties()
+            .get('mozilla.build.kotlin.warnings.as.error', true)
+            .toBoolean()
+    if (enableWarningsAsErrors) {
+        tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).all {
+            kotlinOptions.allWarningsAsErrors = true
+        }
+    } else {
+        println("Warning: `kotlinOptions.allWarningsAsErrors` is not enabled for module ':${project.name}'.")
```

Comment:
`$(project.name)` would resolve to the root project `android-components` right? Is that intentional or did you mean for it to be on each component?

Reply:
`$project` will be resolved relative to `subproject {}`.
To use root project we need `$rootProject` or `$project.rootProject`. :)

Reply:
Oh yes, this is in `allprojects`, thanks!